### PR TITLE
Add flash message translation for log out

### LINKED ISSUE #8628: Missing translation: Successfully logged out message
## Description
When logging out the green confirmation banner at the top of the screen is not translated.

## Expected Behavior
The message should be tranlated.

## Actual Behaviour
It is not.

## Steps to Reproduce
1. Log in
2. Select a language other than English
3. Log out
4. Observe the confirmation banner

## Animated Gif/Screenshot
Sign out

## Workaround
Learn English.

## Severity
bug-s3: a feature is broken but there is a workaround

## Your Environment
* Version used: v4.1.10
* Browser name and version: Firefox 95
* Operating System and version (desktop or mobile): Ubuntu 20.04 LTS Desktop

### PR DESCRIPTION
**Pull request by SarvarKh**
#### What? Why?

Closes #8628

Add missing translation on flash: Successfully logged out message

#### What should we test?

1. Log in
2. change website language other than English
3. Log out => flash message shall be displayed in the selected langugage 

#### Release notes

Added log out flash message translation

Changelog Category: User facing changes

### CODE REVIEW

File: app/controllers/spree/user_sessions_controller.rb [hunk 1]
```diff
@@ -45,6 +45,7 @@ def destroy
       # Logout will clear session data including shopfront_redirect
       #   Here we store it before actually logging out so that the redirect works correctly
       @shopfront_redirect = session[:shopfront_redirect]
+      flash.notice = t('devise.sessions.signed_out')
```
Comment (Matt-Yorkley) [inline app/controllers/spree/user_sessions_controller.rb:48]: Devise already adds the flash message, we shouldn't be adding one here.

Can we drop this line?

File: config/locales/en.yml [hunk 2]
```diff
@@ -159,6 +159,8 @@ en:
       already_registered: "This email address is already registered. Please log in to continue, or go back and use another email address."
     success:
       logged_in_succesfully: "Logged in successfully"
+    sessions:
+      signed_out: "Signed out successfully."
```
Comment (Matt-Yorkley) [inline config/locales/en.yml:163]: I see there's a similar key in Devise's translation file here and it's used if a logout request is received when the user is already logged out. We may as well add it here.

```suggestion
      signed_out: "Signed out successfully."
      already_signed_out: "Signed out successfully."      
```

**Review by Matt-Yorkley — COMMENTED** [review]
It would be good to include a test here. We can add it in `spec/controllers/spree/user_sessions_controller_spec.rb` near the bottom and it'll look a bit like this:

```ruby
describe "#destroy" do
  it "redirects to root with flash message" do
    spree_post :destroy

    expect(response).to redirect_to root_path
    expect(flash[:notice]).to eq "Signed out successfully."
  end
end
```

File: app/jobs/order_cycle_closing_job.rb [hunk 3]
```diff
@@ -1,6 +1,6 @@
 # frozen_string_literal: true
 
-class OrderCycleClosingJob < ActiveJob::Base
+class OrderCycleClosingJob < ApplicationJob
```
Comment (Matt-Yorkley) [inline app/jobs/order_cycle_closing_job.rb:3]: We should probably drop this for now, as we don't currently have an `ApplicationJob` class. You can ignore the linter in this case :+1:

Comment (SarvarKh) [reply app/jobs/order_cycle_closing_job.rb]: Thank you for your advice @Matt-Yorkley 
I just removed it. And could you give me a few minutes to squash my commits, I assume I will have to do so...🙏

**Review by Matt-Yorkley — APPROVED** [review]
:+1:

**Review by apricot12 — APPROVED** [review]
LGTM 👍

**Comment by SarvarKh** [comment]
Hi @Gitanejo, seems like I don't have access to those links (I had requested it via google docs's page). If there is any conflicts/comments, I would work over them🙂

**Comment by Gitanejo** [comment]
> 

Hey @SarvarKh I have granted you access. Thanks!

**Comment by SarvarKh** [comment]
Looks like everything is OK.

File: spec/controllers/spree/user_sessions_controller_spec.rb [hunk 4]
```diff
@@ -30,4 +30,13 @@
       end
     end
   end
+
+  describe "destroy" do
+    it "redirects to root with flash message" do
+      spree_post :destroy
+
+      expect(response).to redirect_to root_path
+      expect(flash[:notice]).to eq "Signed out successfully."
```
Comment (filipefurtad0) [inline spec/controllers/spree/user_sessions_controller_spec.rb:39]: Hey @SarvarKh ,
I'm wondering, which flash notice is this line testing exactly?

I was expecting to find something like `[:flash] = t('devise.sessions.signed_out')` somewhere in the code or in this PR...

The reason I'm asking is that I could not get the translation to work on this PR, neither on a staging server nor locally, after adding the respective keys for the different locales.

Maybe everything is fine with the PR, just I could not observe that it is closing the issue. Thanks for your help on this.

Comment (Matt-Yorkley) [reply spec/controllers/spree/user_sessions_controller_spec.rb:39]: @filipefurtad0 The code which sets those messages is tucked away in the Devise gem, it doesn't live in our codebase. Did you restart puma after changing the translations locally?

Comment (filipefurtad0) [reply spec/controllers/spree/user_sessions_controller_spec.rb:39]: Ah I see, thanks Matt.

I did restart puma with `bundle exec pumactl -F /home/openfoodnetwork/apps/openfoodnetwork/shared/config/puma.rb start` on the staging server, and locally also stopped and restarted the rails server, after adding those keys to `en_CA.yml` and `fr_CA.yml` on staging UK.

Observed the change while using the app in English, but selecting French basically displayed `en_CA.yml`, which is the default language on the server. So yeah, I was wondering if something was missing.

But yeah, nothing breaks so I guess we can merge and check again if it works in production?

Comment (Matt-Yorkley) [reply spec/controllers/spree/user_sessions_controller_spec.rb:39]: I did a quick dev-test and changed those keys in `en.yml` and saw the changes as expected, so it's definitely using them :shrug: 

I think we can merge it :+1:

Comment (filipefurtad0) [reply spec/controllers/spree/user_sessions_controller_spec.rb:39]: Cool, thanks for double checking!

Comment (SarvarKh) [reply spec/controllers/spree/user_sessions_controller_spec.rb:39]: Thanks @filipefurtad0 and @Matt-Yorkley 🙏

**Pull request merged**